test/nudge_web/controllers/site_controller_test.exs

```diff
@@ -0,0 +1,55 @@
+defmodule NudgeWeb.SiteControllerTest do
+  use NudgeWeb.ConnCase
+  use Plug.Test
+  @create_attrs %{active: true, tz: "12.30", url: "https://example.com"}
+  @invalid_attrs %{active: nil, tz: nil, url: nil}
+
+  def fixture(:site) do
+    {:ok, site} = Nudge.Accounts.create_site(@create_attrs)
+    site
+  end
+
+  def fixture(:user) do
+    user_attrs = %{
+      "email" => "abc@email.com", "password" => "passworddd", "name" => "name"
+    }
+    {:ok, user} = Nudge.Accounts.create_user(user_attrs)
+    user
+  end
+
+  describe "create site" do
+    setup [:create_user]
+
+    test "redirects to show when data is valid", %{conn: conn, user: user} do
+      conn = Plug.Test.init_test_session(conn, current_user_id: user.id)
+      conn = post(conn, Routes.site_path(conn, :create), site: @create_attrs)
+
+      assert %{id: id} = redirected_params(conn)
+      assert redirected_to(conn) == Routes.site_path(conn, :show, id)
+
+      conn = get(conn, Routes.site_path(conn, :show, id))
+      assert html_response(conn, 200) =~ "Your Site"
```

Comment:
this is not part of this test case. you said the test is "redirects to show when data is valid". 
"show" itself is another request, and needs to be on another test.